src/coreclr/jit/morph.cpp

```diff
@@ -3001,12 +3004,9 @@ void CallArgs::AddFinalArgsAndDetermineABIInfo(Compiler* comp, GenTreeCall* call
 #endif
         }
 
-        if (GlobalJitOptions::compFeatureHfa)
```

Comment:
Are we relying on `varTypeIsValidHfaType` to report `false` if the feature isn't supported?

-- I'm also not overly familiar with the HFA handling. Does that also include HVA (that is ensuring `Vector128<T>`, which is `__m128`, and up to 4 sequential `Vector128<T>` are passed in register)?

Reply:
> Are we relying on `varTypeIsValidHfaType` to report false if the feature isn't supported?

Yep. It is "the pattern" for HFAs in the compiler.

> Does that also include HVA?

Also yes, for them the "HFA element type" will be `SIMD` (see `HfaTypeFromElemKind`).